src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java

```diff
@@ -268,29 +269,54 @@ private static int getCurrentOrFollowingIndelLength(final PileupElement pe) {
         return pe.isDeletion() ? pe.getCurrentCigarElement().getLength() : pe.getLengthOfImmediatelyFollowingIndel();
     }
 
-    private static double indelQual(final int indelLength) {
-        return INDEL_START_QUAL + (indelLength - 1) * INDEL_CONTINUATION_QUAL;
+    private static byte indelQual(final int indelLength) {
+        return (byte) Math.min(INDEL_START_QUAL + (indelLength - 1) * INDEL_CONTINUATION_QUAL, Byte.MAX_VALUE);
     }
 
-    private static Pair<Integer, Double> altCountAndQualSum(final ReadPileup pileup, final byte refBase) {
-        int altCount = 0;
-        double qualSum = 0;
+    private static List<Byte> altQuals(final ReadPileup pileup, final byte refBase) {
+        final List<Byte> result = new ArrayList<>();
 
         for (final PileupElement pe : pileup) {
             final int indelLength = getCurrentOrFollowingIndelLength(pe);
             if (indelLength > 0) {
-                altCount++;
-                qualSum += indelQual(indelLength);
+                result.add(indelQual(indelLength));
             } else if (isNextToUsefulSoftClip(pe)) {
-                altCount++;
-                qualSum += indelQual(1);
+                result.add(indelQual(1));
             } else if (pe.getBase() != refBase && pe.getQual() > MINIMUM_BASE_QUALITY) {
-                altCount++;
-                qualSum += pe.getQual();
+                result.add(pe.getQual());
             }
         }
 
-        return new Pair<>(altCount, qualSum);
+        return result;
+    }
+
+    // this implements the isActive() algorithm described in docs/mutect/mutect.pdf
+    private static double lnLikelihoodRatio(final int refCount, final List<Byte> altQuals) {
+        //TODO: we're only taking integer digammas so if it's expensive we can cache them
+        final double beta = refCount + 1;
+        final double alpha = altQuals.size() + 1;
+        final double digammaAlpha = Gamma.digamma(alpha);
+        final double digammaBeta = Gamma.digamma(beta);
+        final double digammaAlphaPlusBeta = Gamma.digamma(alpha + beta);
+        final double rho = digammaBeta - digammaAlphaPlusBeta;
+        final double tau = digammaAlpha - digammaAlphaPlusBeta;
+
+
+
+        final double betaEntropy = Beta.logBeta(alpha, beta) - (alpha - 1)*digammaAlpha - (beta-1)*digammaBeta + (alpha + beta - 2)*digammaAlphaPlusBeta;
+
+        // TODO: check if the stream is too expensive
+        final double result = -betaEntropy + rho * refCount + altQuals.stream().mapToDouble(qual -> {
```

Comment:
As I said above, I think we want to add entropies here

Reply:
done